mailbox/event/event-cassandra/src/main/java/org/apache/james/mailbox/events/CassandraEventDeadLetters.java

```diff
@@ -29,26 +29,34 @@
 public class CassandraEventDeadLetters implements EventDeadLetters {
 
     private final CassandraEventDeadLettersDAO cassandraEventDeadLettersDAO;
+    private final CassandraEventDeadLettersGroupDAO cassandraEventDeadLettersGroupDAO;
 
     @Inject
-    public CassandraEventDeadLetters(CassandraEventDeadLettersDAO cassandraEventDeadLettersDAO) {
+    public CassandraEventDeadLetters(CassandraEventDeadLettersDAO cassandraEventDeadLettersDAO,
+                                     CassandraEventDeadLettersGroupDAO cassandraEventDeadLettersGroupDAO) {
         this.cassandraEventDeadLettersDAO = cassandraEventDeadLettersDAO;
+        this.cassandraEventDeadLettersGroupDAO = cassandraEventDeadLettersGroupDAO;
     }
 
     @Override
     public Mono<Void> store(Group registeredGroup, Event failDeliveredEvent) {
         Preconditions.checkArgument(registeredGroup != null, REGISTERED_GROUP_CANNOT_BE_NULL);
         Preconditions.checkArgument(failDeliveredEvent != null, FAIL_DELIVERED_EVENT_CANNOT_BE_NULL);
 
-        return cassandraEventDeadLettersDAO.store(registeredGroup, failDeliveredEvent);
+        return cassandraEventDeadLettersDAO.store(registeredGroup, failDeliveredEvent)
+            .then(cassandraEventDeadLettersGroupDAO.storeGroup(registeredGroup));
     }
 
     @Override
     public Mono<Void> remove(Group registeredGroup, Event.EventId failDeliveredEventId) {
         Preconditions.checkArgument(registeredGroup != null, REGISTERED_GROUP_CANNOT_BE_NULL);
         Preconditions.checkArgument(failDeliveredEventId != null, FAIL_DELIVERED_ID_EVENT_CANNOT_BE_NULL);
 
-        return cassandraEventDeadLettersDAO.removeEvent(registeredGroup, failDeliveredEventId);
+        return cassandraEventDeadLettersDAO.removeEvent(registeredGroup, failDeliveredEventId)
+            .then(failedEventIds(registeredGroup)
```

Comment:
Following RMK-1 we don't need this expensive operation to happen.

Note also that this is not concurrently friendly